src/main/java/com/artipie/front/settings/YamlCredentials.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2022 artipie.com
+ * https://github.com/artipie/front/LICENSE.txt
+ */
+package com.artipie.front.settings;
+
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.artipie.front.auth.Credentials;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.apache.commons.codec.binary.Hex;
+
+/**
+ * Yaml credentials parser.
+ * @since 1.0
+ */
+public final class YamlCredentials implements Credentials {
+
+    /**
+     * Yaml source.
+     */
+    private final YamlMapping mapping;
+
+    /**
+     * New yaml credentials.
+     * @param mapping Yaml
+     */
+    public YamlCredentials(final YamlMapping mapping) {
+        this.mapping = mapping;
+    }
+
+    @Override
+    public Optional<Credentials.User> user(final String name) {
+        return Optional.ofNullable(this.mapping.yamlMapping("credentials"))
+            .flatMap(cred -> Optional.ofNullable(cred.yamlMapping(name)))
+            .map(User::new);
+    }
+
+    /**
+     * Yaml user item.
+     * @since 1.0
+     */
+    private static final class User implements Credentials.User {
+
+        /**
+         * Yaml source.
+         */
+        private final YamlMapping mapping;
+
+        /**
+         * New user.
+         * @param mapping Yaml
+         */
+        private User(final YamlMapping mapping) {
+            this.mapping = mapping;
+        }
+
+        // @checkstyle ReturnCountCheck (30 lines)
+        @Override
+        @SuppressWarnings("PMD.OnlyOneReturn")
+        public boolean validatePassword(final String pass) {
+            final var config = this.mapping.string("pass");
+            if (config == null) {
+                throw new IllegalStateException(
+                    "invalid credentials configuration: `pass` field not found"
+                );
+            }
+            final var parts = config.split(":");
```

Comment:
we actually support two formats for password and its type now in artipie, the one which is implemented here and another, where password type is its own mapping:
```
username:
  type: plain
  pass: 123
```

check the [doc](https://github.com/artipie/artipie#multitenancy) and the [code](https://github.com/artipie/artipie/blob/5ee169b9ec8f6c32436c5e0c647fb022b00d5896/src/main/java/com/artipie/auth/AuthFromYaml.java#L54)